src/test/frontend/deploy/deployfromsettings_controller_test.js

```diff
@@ -336,4 +336,85 @@ describe('DeployFromSettings controller', () => {
       expect(ctrl.imagePullSecret).toEqual('');
     });
   });
+
+  /**
+   * The value entered for ‘App Name” is used implicitly as the name for several resources (pod, rc,
+   * svc, label). Therefore, the app-name validation pattern is based on the RC-pattern, but must
+   * conform with all validation patterns of all the created resources.
+   * Currently, the ui pattern that conforms with all patterns is alpha-numeric with dashes between.
+   */
+
+  it('should allow strings that conform to the patterns of all created resources', () => {
+    // given the pattern used by the App Name field in the UI
+    let namePattern = ctrl.namePattern;
+    // given the patterns of all the names that are implicitly created
+    let allPatterns = {
+      servicePattern: "[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*",
+      labelPattern: "(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?",
+      rcPattern: "[a-z0-9]([-a-z0-9]*[a-z0-9])?",
+      appNamePattern: namePattern,
+    };
+
+    // then
+    for (let pattern in allPatterns) {
+      expect("mylowercasename".match(allPatterns[pattern])).toBeDefined();
+      expect("my-name-with-dashes-between".match(allPatterns[pattern])).toBeDefined();
+      expect("my-n4m3-with-numb3r5".match(allPatterns[pattern])).toBeDefined();
+    }
+  });
+
+  /**
+   * The value entered for 'App Name' is used implicitly as the name for several resources (pod, rc,
+   * svc, label).
+   * Remark: The app-name pattern excludes some service names and label values, which could be
+   * created manually. This is a restriction of the current design.
+   */
+  it('should reject names that fail to conform to appNamePattern', () => {
+
+    // given
+    let appNamePattern = ctrl.namePattern;
+
+    // then the following valid service names will be rejected
+    expect("validservice.com".match(appNamePattern)).toBeNull();
```

Comment:
Btw, use single quotes everywhere `'`. This is a pattern in the project, and generally in JavaScript. `"` gets bad when you do this in HTML.